### PR TITLE
Add Create/Update RPCs for Auth Connectors

### PR DESCRIPTION
**Pull request by rosstimothy**
This is phase 1 of  adding optimistic locking support to auth connectors. Server side changes will land in follow up PRs to prevent breaking e.

Contributes to #30416.

### CODE REVIEW

**Review by strideynet — APPROVED** [review]

File: api/client/client.go [hunk 1]
```diff
@@ -1670,13 +1670,39 @@ func (c *Client) GetOIDCConnectors(ctx context.Context, withSecrets bool) ([]typ
...
 	connector, ok := oidcConnector.(*types.OIDCConnectorV3)
 	if !ok {
 		return trace.BadParameter("invalid type %T", oidcConnector)
 	}
-	_, err := c.grpc.UpsertOIDCConnector(ctx, connector)
+
+	_, err := c.grpc.UpsertOIDCConnectorV2(ctx, &proto.UpsertOIDCConnectorRequest{Connector: connector})
+	// TODO(tross) DELETE IN 16.0.0
+	if err != nil && trace.IsNotImplemented(err) {
+		_, err := c.grpc.UpsertOIDCConnector(ctx, connector) //nolint:staticcheck // SA1019. Kept for backward compatibility.
```
Comment (espadolini) [inline api/client/client.go:1703]: The comment makes it seem like we're only ignoring staticcheck's SA1019 even though the whole staticcheck linter is disabled (golangci-lint doesn't support the staticcheck-specific comments 😢).
```suggestion
		_, err := c.grpc.UpsertOIDCConnector(ctx, connector) //nolint:staticcheck // Deprecated RPC used for backward compatibility.
```

File: api/client/client.go [hunk 2]
```diff
@@ -1735,13 +1761,39 @@ func (c *Client) GetSAMLConnectors(ctx context.Context, withSecrets bool) ([]typ
...
+	samlConnector, ok := connector.(*types.SAMLConnectorV2)
 	if !ok {
 		return trace.BadParameter("invalid type %T", connector)
 	}
-	_, err := c.grpc.UpsertSAMLConnector(ctx, samlConnectorV2)
+
+	_, err := c.grpc.UpsertSAMLConnectorV2(ctx, &proto.UpsertSAMLConnectorRequest{Connector: samlConnector})
+	// TODO(tross) DELETE IN 16.0.0
+	if err != nil && trace.IsNotImplemented(err) {
+		_, err := c.grpc.UpsertSAMLConnector(ctx, samlConnector) ///nolint:staticcheck // SA1019. Kept for backward compatibility.
```
Comment (espadolini) [inline api/client/client.go:1794]:
```suggestion
		_, err := c.grpc.UpsertSAMLConnector(ctx, samlConnector) //nolint:staticcheck // Deprecated RPC used for backward compatibility.
```

File: api/client/client.go [hunk 3]
```diff
@@ -1800,13 +1852,38 @@ func (c *Client) GetGithubConnectors(ctx context.Context, withSecrets bool) ([]t
...
 func (c *Client) UpsertGithubConnector(ctx context.Context, connector types.GithubConnector) error {
 	githubConnector, ok := connector.(*types.GithubConnectorV3)
 	if !ok {
 		return trace.BadParameter("invalid type %T", connector)
 	}
-	_, err := c.grpc.UpsertGithubConnector(ctx, githubConnector)
+	_, err := c.grpc.UpsertGithubConnectorV2(ctx, &proto.UpsertGithubConnectorRequest{Connector: githubConnector})
+	// TODO(tross) DELETE IN 16.0.0
+	if err != nil && trace.IsNotImplemented(err) {
+		_, err := c.grpc.UpsertGithubConnector(ctx, githubConnector) ///nolint:staticcheck // SA1019. Kept for backward compatibility testing.
```
Comment (espadolini) [inline api/client/client.go:1884]:
```suggestion
		_, err := c.grpc.UpsertGithubConnector(ctx, githubConnector) //nolint:staticcheck // Deprecated RPC used for backward compatibility.
```

**Review by espadolini — APPROVED** [review]
What's the plan for the item-returning upsert operation?

**Pull request merged**